tests/ui/impl-trait/ice-unexpected-param-type-whensubstituting-in-region-112823.rs

```diff
@@ -26,9 +26,11 @@ impl X for Y {
     //~^ ERROR type `LineStream` has 0 type parameters but its trait declaration has 1 type parameter
     type LineStreamFut<'a, Repr> = impl Future<Output = Self::LineStream<'a, Repr>>;
     fn line_stream<'a, Repr>(&'a self) -> Self::LineStreamFut<'a, Repr> {}
```

Comment:
This error is a bit iffy, ideally it would point to the error when failing to prove `AliasRelate(impl Future<Output = Self::LineStream<'a, Repr>>, ())`, However, this is a nested goal with `GoalSource::Misc` of an `NormalizesTo` goal, so we currently do not recurse into it in the `BestObligations` visitor.

Think that we should probably look at `Misc` goals if all non-misc goals fail or something. This requires some experimentation so I would leave it for a separate PR.

Reply:
I'd rather we figure out how to mark specific goals that are currently marked as misc today. Looking into *all* misc goals seems way too broad.